core/src/main/scala/org/apache/spark/executor/MesosExecutorBackend.scala

```diff
@@ -81,6 +90,12 @@ private[spark] class MesosExecutorBackend
       env)
   }
 
+  private def getRangesFromResources(executorInfo:ExecutorInfo): List[(Long, Long)] ={
+  executorInfo.getResourcesList.asScala.filter(r => r.getType == Value.Type.RANGES & r.getName == "ports").toList.
+      map(r => r.getRanges.getRangeList).flatMap { ranges => ranges.asScala.map { r => (r.getBegin, r.getEnd) }}
+  }
```

Comment:
This is taken from other part of the code base it is actually safer. I just reused it form elsewhere.